client/src/components/CommonComponents/CharityCount/index.js

```diff
@@ -1,52 +1,58 @@
+/* eslint-disable jsx-a11y/no-static-element-interactions */
+/* eslint-disable jsx-a11y/click-events-have-key-events */
 /* eslint-disable react/sort-comp */
 /* eslint-disable react/prop-types */
-import React, { Component } from 'react';
+import React from 'react';
 import { Link } from 'react-router-dom';
+import { CountContext } from '../../Context/CountContext';
 
 import './style.css';
 
-class CharityCount extends Component {
-  state = {
-    CharityList: [],
-  };
-
-  static getDerivedStateFromProps() {
-    const CharityList = JSON.parse(localStorage.getItem('listCharity'));
-    return { CharityList };
-  }
-
-  render() {
-    const { CharityList } = this.state;
-    const select = CharityList || [];
-    const count = select.length;
-    let str = '/compare?';
-    str += `${select.map((x, i) => `charity${i + 1}=${x}`).join('&')}`;
-    return (
-      <Link to={count > 1 && str}>
+const CharityCount = () => (
+  <CountContext.Consumer>
+    {props => {
+      const { charityList, removeCharity } = props;
+      const count = charityList.length;
+      const str = `/compare?${charityList
+        .map((charity, i) => `charity${i + 1}=${charity.charityId}`)
+        .join('&')}`;
+      return (
         <div className="count-charity">
           <div className="tooltip">
-            <img
-              className="img-charity"
-              src="http://www7.0zz0.com/2018/12/05/10/361893112.png"
-              alt="img charity"
-            />
-            <div className="count">
-              <strong>{count}</strong>
-            </div>
+            <Link to={count > 1 && str}>
+              <img
+                className="img-charity"
+                src="http://www7.0zz0.com/2018/12/05/10/361893112.png"
```

Comment:
@asmaa-ezz I know this line was moved, not added, in this PR, but I have a quick question - is there any chance that the image hosted on 0zz0.com could disappear on you?  It might be a good idea to copy that image into your repo, if that's allowed, and host it yourself.  Then there's no chance that this image breaks in the future.